src/View/Main/Main.js

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useState } from 'react';
+import List from '../../Components/List/List';
+import Controls from '../../Controls/Controls';
+import Page from '../../Controls/Page/Page';
+import { fetchPokemon, fetchPokemonTypes } from '../../services/fetchpoke';
+import './Main.css';
+
+export default function Main() {
+  const [data, setData] = useState([]);
+  const [types, setTypes] = useState([]);
+  const [searchKey, setSearchKey] = useState('pokemon');
+  const [sortKey, setSortKey] = useState('pokemon');
+  const [selectedType, setSelectedType] = useState('All');
+  const [query, setQuery] = useState('');
+  const [debouncedQuery, setDebouncedQuery] = useState(query);
+  const [order, setOrder] = useState('asc');
+  const [page, setPage] = useState(1);
+  const [count, setCount] = useState(0);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetch = async () => {
```

Comment:
nice work on utilizing a single useEffect to manage your state. Understanding the dependency array is super powerful within React.